dipy/testing/decorators.py

```diff
@@ -103,3 +110,111 @@ def _set_random_number_generator_wrapper(pytestconfig, *args, **kwargs):
         return _set_random_number_generator_wrapper
 
     return _set_random_number_generator
+
+
+def warning_for_keywords(from_version, until_version):
+    """
+    Decorator to warn about keyword arguments passed as positional arguments
+    and handle version-based deprecation of functions.
+
+    Parameters
+    ----------
+    from_version : str
+        The version from which the warning should start.
+    until_version : str
+        The version until which the warning is applicable.
+
+    Returns
+    -------
+    function
+        The wrapped function that will issue a warning or raise an error based
+        on the version.
+
+    Raises
+    ------
+    RuntimeError
+        If the current version is greater than or equal to `until_version`.
+    """
+    
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            current_version = dipy.__version__
+
+            def convert_positional_to_keyword(func, args, kwargs):
+                """
+                Converts excess positional arguments to keyword arguments.
+                
+                Parameters
+                ----------
+                func : function
+                    The original function to be called.
+                args : tuple
+                    The positional arguments passed to the function.
+                kwargs : dict
+                    The keyword arguments passed to the function.
+
+                Returns
+                -------
+                result
+                    The result of the function call with corrected arguments.
+                """
+
+                sig = signature(func)
+                params = sig.parameters
+                max_positional_args = sum(
+                    1
+                    for param in params.values()
+                    if param.kind in
+                    (Parameter.POSITIONAL_ONLY, Parameter.POSITIONAL_OR_KEYWORD)
+                )
+
+                if len(args) > max_positional_args:
+                    # Split positional arguments into valid positional and 
+                    # those needingconversion
```

Comment:
I think that the CodeSpell CI job will give an error here.
`needingconversion` -> `needing conversion`

Reply:
Okay, the CodeSpell does not complain, sorry, I did not check the job status.

However, Ruff does complain about some code formatting. Check it when you can please.

Also, there is a problem with the 'Optional Deps' job for MacOS, Python version 3.10. I am not an expert in CI related problems, maybe @skoudoro can give you a hint.

Reply:
understood. Thanks